l10n_th_address/__init__.py

```diff
@@ -2,3 +2,14 @@
 # © 2017 Ecosoft (ecosoft.co.th).
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 from . import models
+from odoo import api, SUPERUSER_ID
+
+
+def _update_data_noupdate(cr, registry):
+    cr.execute("""
+        update ir_model_data
+        set noupdate = true
+        where model in ('res.country.province',
+                        'res.country.district',
+                        'res.country.township')
+    """)
```

Comment:
One may imagine situations where res.country.province is declared and populated in another module. This may happen when you install several localization at once. In this case, this update would modify data that belong to other modules. I think you can add `where module= ` to limit to your own module data.